File: 3275-minimum-number-of-pushes-to-type-word-i/minimum-number-of-pushes-to-type-word-i.py
class Solution:
    def minimumPushes(self, word: str) -> int:
        n = len(word)
        ans = 0
        num = 2
        mp = defaultdict(str)

        for i in word:
            mp[num] += i
            num += 1
            if(num == 10):
                num = 2

        for i in word:
            for j in range(2, 10):
                if i in mp[j]:
                    ans += (mp[j].index(i) + 1)
                    break

        return ans
        

File: 3275-minimum-number-of-pushes-to-type-word-i/minimum-number-of-pushes-to-type-word-i.go
func minimumPushes(word string) int {
	n := len(word)

	if n <= 8 {
		return n
	} else if n <= 16 {
		n -= 8
		return 8 + (n * 2)
	} else if n <= 24 {
		n -= 16
		return 24 + (n * 3)
	} else {
		n -= 24
		return 48 + (n * 4)
	}
}